pkg/platform/factory/factory.go

```diff
@@ -57,15 +62,65 @@ func CreatePlatform(parentLogger logger.Logger,
 		if kubeconfigPath != "" || kube.IsInCluster() {
 
 			// call again, but force kube
-			return CreatePlatform(parentLogger, "kube", platformConfiguration)
-		}
+			newPlatform, err = CreatePlatform(parentLogger, "kube", platformConfiguration, defaultNamespace)
+		} else {
 
-		// call again, force local
-		return CreatePlatform(parentLogger, "local", platformConfiguration)
+			// call again, force local
+			newPlatform, err = CreatePlatform(parentLogger, "local", platformConfiguration, defaultNamespace)
+		}
 
 	default:
 		return nil, fmt.Errorf("Can't create platform - unsupported: %s", platformType)
 	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create %s platform", platformType)
+	}
+
+	if err = ensureDefaultProjectExistence(parentLogger, newPlatform, defaultNamespace); err != nil {
+		return nil, errors.New("Failed to ensure default project existence")
+	}
+
+	return newPlatform, nil
+}
+
+func ensureDefaultProjectExistence(parentLogger logger.Logger, p platform.Platform, defaultNamespace string) error {
+	resolvedNamespace := p.ResolveDefaultNamespace(defaultNamespace)
+
+	projects, err := p.GetProjects(&platform.GetProjectsOptions{
+		Meta: platform.ProjectMeta{
+			Name:      "default",
+			Namespace: resolvedNamespace,
+		},
+	})
+	if err != nil {
+		return errors.New("Failed to get projects")
+	}
+
+	if len(projects) != 1 {
+
+		// if we're here the default project doesn't exist. create it
+		projectConfig := platform.ProjectConfig{
```

Comment:
if by freak accident more then once will be created this will keep creating it on each restart of the pod.
differentiate 0, 1, and > 1 (error out?)

Reply:
It cannot be > 1, k8s should prevent creating more than 1 resource with the same id

Reply:
1. The mapping between project.name and the k8s resource name might change
2. Isn't that code abstract and should take to non k8s platforms?